app/packages/core/src/components/Modal/modal-spaces-utils.ts

```diff
@@ -0,0 +1,82 @@
+import { PluginComponentRegistration } from "@fiftyone/plugins";
+import { SpaceNodeJSON, usePanels } from "@fiftyone/spaces";
+import { panelsCompareFn } from "@fiftyone/spaces/src/utils/sort";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+const MODAL_PLUGINS_REGISRATION_TIMEOUT_MS = 200;
+
+export const SAMPLE_MODAL_PLUGIN_NAME = "fo-sample-modal-plugin";
+const SAMPLE_MODAL_PLUGINS_LOCAL_STORAGE_KEY = "fo-sample-modal-plugins";
+
+export const useModalSpaces = () => {
+  const [modalSpaces, setModalSpaces] = useState<SpaceNodeJSON | null>(null);
+
+  const panelsPredicate = useCallback(
```

Comment:
I think we should make this first-class in `usePanels`. For example

```js

function usePanel(surface: string) {
  return filteredPanelForSurface
}

```

Reply:
Sounds good.